airbyte-integrations/connectors/source-shopify/source_shopify/source.py

```diff
@@ -969,6 +105,15 @@ def test_connection(self) -> tuple[bool, str]:
         except MissingAccessTokenError:
             return False, self.describe_error("missing_token_error")
 
+    def get_shop_id(self) -> str:
+        """
+        We need to have the `shop_id` value available to have it passed elsewhere and fill-in the missing data.
+        By the time this method is tiggered, we are sure we've passed the `Connection Checks` and have the `shop_id` value.
+        """
+        response = list(self.test_stream.read_records(sync_mode=None))
```

Comment:
this looks fishy. sync_mode isn't optional in the read_records implementation

Reply:
corrected.